### PR TITLE
POND-184: fix describe and simplify query compiler interface

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -1165,54 +1167,33 @@ def describe(
...
-            percentiles = list(percentiles)
-
-            # get them all to be in [0, 1]
-            validate_percentile(percentiles)
-
-            # median should always be included
-            if 0.5 not in percentiles:
-                percentiles.append(0.5)
-            percentiles = np.asarray(percentiles)
+        # copied from pandas.core.describe.describe_ndframe
```
Comment (pyrito) [inline modin/pandas/base.py:1170]: that's what I like to see 🤣

File: modin/pandas/base.py [hunk 2]
```diff
@@ -1165,54 +1167,33 @@ def describe(
...
+            data = self.select_dtypes(include=default_include)
+            if len(data.columns) == 0:
+                data = self
+        elif include == "all":
+            if exclude is not None:
+                msg = "exclude must be None when include is 'all'"
+                raise ValueError(msg)
+            data = self
         else:
-            percentiles = np.array([0.25, 0.5, 0.75])
```
Comment (pyrito) [inline modin/pandas/base.py:1210]: will the default percentiles be handled in the qc?

Comment (mvashishtha) [reply modin/pandas/base.py:1210]: `refine_percentiles` handles that : https://github.com/pandas-dev/pandas/blob/7c4b98e7940a1d31bbdbd0011ab3ec027090abb1/pandas/core/methods/describe.py#L378-L379

**Review by pyrito — APPROVED** [review]

File: modin/pandas/base.py (hunk 2 above)
Comment (pyrito) [reply modin/pandas/base.py:1210]: nice - glad that we are using pandas util functions for further sanity

**Pull request merged**